extensions/amp-iframely/0.1/amp-iframely.js

```diff
@@ -0,0 +1,349 @@
+    const me = this;
+    const iframely = this.element.ownerDocument.createElement('iframe');
+    setStyle(iframely, 'border', '0');
+    iframely.setAttribute('allow', this.allow_);
+    this.src_ += this.appendOptions();
+    iframely.src = this.src_;
+    this.applyFillContent(iframely);
+    this.element.appendChild(iframely);
+
+    /**
+     * Handling Iframely messages
+     * @param {window.event.source} event - event to check iframely against
+     * */
+    function receiveMessage(event) {
+      const iframes = me.element.getElementsByTagName('iframe');
+      if (findIframeByContentWindow(iframes, event.source)) {
+        me.handleEvent(me, event);
+      }
+    }
+    window.addEventListener('message', receiveMessage, false);
+    this.iframely_ = iframely;
+    return this.loadPromise(iframely);
+  }
+
+  /**
+   * Handles Iframely events: widget sizing, cancel, decorate cards
+   * @param {AmpIframely} me - instance of an active component
+   * @param {window.event} event - Iframely message with a method to apply
+   * */
+  handleEvent(me, event) {
+    let data = null;
+    try {
+      data = JSON.parse(event.data);
+    } catch (e) {
+      // Do nothing - likely not Iframely message if JSON errs
+    }
+    if (data) {
+      if (data.method === 'resize') {
+        /** Set the size of the card according to the message from Iframely */
+        me./*OK*/ changeHeight(data['height']);
+      }
+      if (data.method === 'setIframelyEmbedData') {
+        /** apply Iframely card styles if present */
+        const media = data.data.media || null;
+        if (media && media.frame_style) {
+          const styles = media.frame_style.split(';');
+          styles.forEach(function(style) {
+            const styleValue = style.split(':')[1];
+            switch (style.split(':')[0]) {
+              case 'border':
+                setStyle(me.element, 'border', styleValue);
+                break;
+              case 'border-radius':
+                setStyle(me.element, 'border-radius', styleValue);
+                break;
+              case 'box-shadow':
+                setStyle(me.element, 'box-shadow', styleValue);
+                break;
+            }
+          });
+        }
+        if (media && media['aspect-ratio']) {
+          let height;
+          const box = me.element.getLayoutBox();
+          if (media['padding-bottom']) {
+            /** Apply height for media with updated "aspect-ratio" and "padding-bottom". */
+            height =
+              box.width / media['aspect-ratio'] + media['padding-bottom'];
+            me./*OK*/ changeHeight(height);
+          } else {
+            height = box.width / media['aspect-ratio'];
+            if (Math.abs(box.height - height) > 1) {
+              /** Apply new height for updated "aspect-ratio". */
+              me./*OK*/ changeHeight(height);
+            }
+          }
+        }
+      }
+      if (data.method === 'cancelWidget') {
```

Comment:
In which cases can this be executed? Removing the component's box entirely causes content-shift.

- Is it only on user interaction?
- Is it on error? If it's on error, then showing the `fallback` element would be preferrable.